web/src/client/software.js

```diff
@@ -217,78 +196,75 @@ class SoftwareBaseClient {
   /**
    * @param {string|undefined} address - D-Bus address; if it is undefined, it uses the system bus.
    */
-  constructor(address = undefined) {
-    this.client = new DBusClient(SOFTWARE_SERVICE, address);
-    this.product = new ProductManager(this.client);
-    this.proxies = {};
+  /**
+   * @param {import("./http").HTTPClient} client - HTTP client.
+   */
+  constructor(client) {
+    this.client = client;
   }
 
   /**
    * Asks the service to reload the repositories metadata
    *
    * @return {Promise<void>}
    */
-  async probe() {
-    const proxy = await this.client.proxy(SOFTWARE_IFACE);
-    return proxy.Probe();
+  probe() {
+    return this.client.post("/software/probe");
   }
 
   /**
    * Returns how much space installation takes on disk
    *
-   * @return {Promise<string>}
+   * @return {Promise<SoftwareProposal>}
    */
-  async getUsedSpace() {
-    const proxy = await this.client.proxy(SOFTWARE_IFACE);
-    return proxy.UsedDiskSpace();
+  getProposal() {
+    return this.client.get("/software/proposal");
   }
 
   /**
    * Returns available patterns
    *
-   * @param {boolean} filter - `true` = filter the patterns, `false` = all patterns
-   * @return {Promise<Array<string>>}
+   * @return {Promise<Pattern[]>}
    */
-  async patterns(filter) {
-    const proxy = await this.client.proxy(SOFTWARE_IFACE);
-    return proxy.ListPatterns(filter);
+  async getPatterns() {
+    /** @type Array<{ name: string, category: string, summary: string, description: string, order: string, icon: string }> */
```

Comment:
Question: Should we define this type elsewhere and the use it here?

Reply:
Good question. I do not think it is needed, as it is the only place it is used. We convert those objects into our `Pattern` specification.